### DISCUSSION TITLE
how to save map using ros

### DISCUSSION THREAD

**Discussion by DavidePatria** [OpenVSLAM-Community/openvslam_ros#47] [Q&A] [answered]
using openvslam stand-alone it is possible to save the generated map in a specific location by simply using `--map-db /path/to/map` , but when using ros the option isn't accepted and ros returns an error.
The thing is that the localization mode still needs a map file, which is provided in the same way.
So, since the file apparently cannot be generated (or found, simply) using ros, does it have to be built using the standalone version?
If so, how does one use a rgbd camera for the job?

**Comment by ymd-stella** [comment, marked as answer]
> when using ros the option isn't accepted and ros returns an error.

Please describe the command and error details. The option exists,

https://github.com/OpenVSLAM-Community/openvslam_ros/blob/03646d5d8fb719e33c1fe535917d8f67224932a3/src/run_slam.cc#L144

so if it doesn't work, it's a bug.

> So, since the file apparently cannot be generated (or found, simply) using ros, does it have to be built using the standalone version?
If so, how does one use a rgbd camera for the job?

If you want to run it in the standalone version, see below.

https://openvslam-community.readthedocs.io/en/latest/example.html#tum-rgbd-dataset

However, I think it is easier to load from bag.

**Reply by DavidePatria** [reply]
I don't know what had happened, but today the maps are correctly saved.
thank you
